app/helpers/application_helper.rb

```diff
@@ -61,7 +61,7 @@ def link_to_add_fields(name, f, association, partial = nil, options = {})
     fields = f.fields_for(association, new_object, :child_index => "new_#{association}") do |builder|
       render((partial.nil? ? association.to_s.singularize + "_fields" : partial), :f => builder)
     end
-    link_to_function(name, ("add_fields(this, \"#{association}\", \"#{escape_javascript(fields)}\")").html_safe, add_html_classes(options, "btn btn-success") )
+    link_to_function(name, ("add_fields('#{options[:target]}', '#{association}', '#{escape_javascript(fields)}')").html_safe, add_html_classes(options, "btn btn-success") )
```

Comment:
This is so that the element to add to can be specified to something other then `this` - I needed this change to make the global parameter to align nicely with the puppet class parameters. (https://github.com/theforeman/foreman/pull/2858/files#diff-b74a9e3cff7e9e6af6614aea6e50e3a5R16)